hail/python/hail/table.py

```diff
@@ -2143,7 +2249,8 @@ def _flatten(prefix, s):
                 return [(k, v) for (f, e) in s.items() for (k, v) in _flatten(prefix + '.' + f, e)]
             else:
                 return [(prefix, s)]
-        t = self.key_by()
+        # unkey but preserve order
+        t = self.order_by(*self.key)
```

Comment:
wait, doesn't key_by() preserve order?

Reply:
No, I was also confused.  In particular, this rule:

```
    case TableKeyBy(TableKeyBy(child, _, _), keys, false) if canRepartition =>
      TableKeyBy(child, keys, false)
```

Reply:
Hmm....

I guess we don't make any guarantees about the relative order within identical keys.

But it does feel a bit weird that:

```
ht = ht.key_by('locus')
ht = ht.key_by()
```

will ignore the first key_by

Reply:
I think it makes sense now.  `t.key_by(k)` says make sure the result is keyed by `k`, but nothing more.  In particular, there is no guarantee of stability.

Reply:
yeah, I think it's fine.

I imagine this is a bug in other places, though.

Reply:
we will only guarantee order after an order_by, right?

Reply:
No, we guarantee the order for keyed things, too.  (Although we might stop that in 0.3 when we have a richer set of partitioners and keys should just be for join semantics.)